goas/atomic_asset_transfer/risc0_proofs/zone_state/src/main.rs

```diff
@@ -100,29 +98,36 @@ fn deposit(
 }
 
 fn validate_zone_transition(
-    in_note: &cl::PartialTxInputWitness,
-    out_note: &cl::PartialTxOutputWitness,
-    in_state_cm: &StateCommitment,
-    out_state_cm: &StateCommitment,
+    in_note: cl::PartialTxInputWitness,
+    out_note: cl::PartialTxOutputWitness,
+    meta: ZoneMetadata,
```

Comment:
I think we also need to check the zone metadata is the one used in the input state commitment, if we take the input state witness as input it already contains the metadata and we can then check the commitment matches the input note state

Reply:
this `meta` is actually pulled from the input state, by the caller

I did it this way to avoid having to clone the input state and keeping two copies of the state in memory.

Reply:
I could rename this field to `in_meta`? to make the connection more clear?

Reply:
did that ^

Reply:
sorry missed that